devito/logger.py

```diff
@@ -77,7 +77,7 @@ def set_log_level(level, comm=None):
     """
     from devito import configuration
 
-    if comm is not None:
+    if comm is not None and configuration['mpi']:
```

Comment:
why is this addition needed?

Reply:
To elegantly alleviate the case of DEVITO_MPI=0 mpirun -n 2 (running copies of the problem) as discussed